### PR TITLE
Add NYCCoin contracts

### PR DESCRIPTION
**Pull request by LNow**
NYCCoin contracts deployer: `SP2H8PY27SEZ03MWRKS5XABZYQN17ETGQS3527SA5`
NYC city wallet: `SM18VBF2QYAAHN57Q28E2HSM15F6078JZYZ2FQBCX`

We gonna reuse all traits and VRF contract (all deployed by `SP466FNC0P7JWTNM2R9T199QRZN1MYEDTAR0KP27`)

Approvers wallet addresses remain unchanged.

### CODE REVIEW

**Comment by FriendsFerdinand** [comment]
We're not using v2 for NYC?

**Comment by LNow** [comment]
@FriendsFerdinand - no, V2 is not ready. This is V1 with just tiny modifications which you can check here: https://github.com/citycoins/citycoin/projects/2

We are going to deploy them under `v1` name instead of `v1.1` because it will be first version for NYC.

File: contracts/NewYorkCityCoin/core-v1.clar [hunk 1]
```diff
@@ -0,0 +1,909 @@
+;; NEWYORKCITYCOIN CORE CONTRACT
```
Comment (whoabuddy) [inline contracts/NewYorkCityCoin/core-v1.clar:1]: Same here.

```suggestion
;; NEWYORKCITYCOIN CORE CONTRACT
;; CityCoins Protocol Version 1.0.1
```

File: contracts/NewYorkCityCoin/token.clar [hunk 2]
```diff
@@ -0,0 +1,176 @@
+;; NEWYORKCITYCOIN TOKEN CONTRACT
```
Comment (whoabuddy) [inline contracts/NewYorkCityCoin/token.clar:1]: Same again.

```suggestion
;; NEWYORKCITYCOIN TOKEN CONTRACT
;; CityCoins Protocol Version 1.0.1
```

**Review by whoabuddy — CHANGES_REQUESTED** [review]
Overall this LGTM, great job! One small note about adding the protocol version to this and future contracts in the comments. After this contract is deployed we should tag the `main` branch with the same version number.

**Review by whoabuddy — APPROVED** [review]

**Pull request merged**